### DISCUSSION TITLE
Simple samples for dummies

### DISCUSSION THREAD

**Discussion by Berad** [App-vNext/Polly#1967] [Q&A] [answered]
Hi there!
I've read introduction, get started, etc. Still I can't find any dummy samples.

In my case I need to retry getting some value from my database, i.e. now I have

`MyCustomType myValue = await dbContext.TryGetMyValueByIdAsync(id);`

where `TryGetMyValueByIdAsync` can return null and it's ok after several attempts.
And I was looking for something like

```
MyCustomType myValue = pipeline.ExecuteAsync<MyCustomType>( token => {
    dbContext.TryGetMyValueByIdAsync(id, token)); 
});
```

I don't need to get any exceptions, I just want to retry to get a value several times and if it's null -- ok, I will deal with it.

Could you please help me with that?

**Comment by martincostello** [comment, marked as answer]
For your use case it looks like you want a [fallback strategy](https://www.pollydocs.org/strategies/fallback.html) with something _similar_ to:

```csharp
var options = new FallbackStrategyOptions<MyCustomType>
{
    ShouldHandle = new PredicateBuilder<MyCustomType>().Handle<Exception>(),
    FallbackAction = static args => Outcome.FromResultAsValueTask((MyCustomType)null)
};

var pipeline = new ResiliencePipelineBuilder<MyCustomType>().AddFallback(options);

MyCustomType myValue = pipeline.ExecuteAsync<MyCustomType>(
    token => dbContext.TryGetMyValueByIdAsync(id, token));
```

**Reply by Berad** [reply]
Almost! 
Thank you for your answer, you helped me to realize that I actually need a combination of fallback and retry!
I found it on the page you've linked. Very appreciated!
